Source/WTF/wtf/ThreadGroup.h

```diff
@@ -34,15 +34,15 @@ namespace WTF {
 
 enum class ThreadGroupAddResult { NewlyAdded, AlreadyAdded, NotAdded };
 
-class ThreadGroup final : public std::enable_shared_from_this<ThreadGroup> {
+class ThreadGroup final : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<ThreadGroup> {
     WTF_MAKE_FAST_ALLOCATED;
     WTF_MAKE_NONCOPYABLE(ThreadGroup);
 public:
     friend class Thread;
 
-    static std::shared_ptr<ThreadGroup> create()
+    static Ref<ThreadGroup> create()
     {
-        return std::allocate_shared<ThreadGroup>(FastAllocator<ThreadGroup>());
+        return adoptRef(*new ThreadGroup());
```

Comment:
I would take out the parentheses after ThreadGroup. Also, should move this to the .cpp file since this is probably not good to inline. We’d want to inline the constructor instead.

Reply:
OK, changed.